File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        mostWater = 0
        left, right = 0, len(height)-1

        while left < right:
            if height[left] <= height[right]:
                mostWater = max(mostWater, height[left]*(right-left))
                left += 1
                continue
            mostWater = max(mostWater, height[right]*(right-left))
            right -= 1
        
        return mostWater
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size()-1;
        int maxVolume = INT_MIN;
        while(left < right){
            int currentVolume = (right-left)* (int)min(height[left], height[right]);
            maxVolume = max(maxVolume, currentVolume);
            if(height[left] <= height[right]) left++;
            else right--;
        }
        return maxVolume;
    }
};